app/src/main/java/com/zulip/android/ZulipActivity.java

```diff
@@ -54,6 +57,48 @@
         MessageListFragment.Listener {
 
     ZulipApp app;
+    List<Message> muted_topics;
+    @Override
+    public void addToList(Message message) {
+        muted_topics.add(message);
+    }
+
+    @Override
+    public void unMuteTopic(int StreamID, String subject) {
+        app.UnMuteTopic(StreamID, subject);
+        int added = 0;
+        for (Iterator<Message> messageIterator = muted_topics.iterator(); messageIterator.hasNext(); ) {
+            Message message = messageIterator.next();
+            if (message.getStream().getId() == StreamID && message.getSubject().equals(subject)) {
+                messageIterator.remove();
+                homeList.adapter.add(message);
+                added++;
+            }
+        }
+        if (added > 0) {
+            Collections.sort(homeList.messageList, new Comparator<Message>() {
+                @Override
+                public int compare(Message message1, Message message2) {
+                    return message1.getTimestamp().compareTo(message2.getTimestamp());
+                }
+            });
+            homeList.adapter.notifyDataSetChanged();
+        }
+    }
+
+    @Override
+    public void MuteTopic(int StreamID, String subject) {
+        app.MuteTopic(StreamID, subject);
+        for (int i = homeList.adapter.getCount() - 1; i >= 0; i--) {
```

Comment:
depending on how large the `homeList`, this might be a bit of an expensive operation. it might be worthwhile to add some type of indexed look up for message topics to the `homeList` adapter.  not strictly necessary for this PR tho :)